### ISSUE TITLE
nested repeat nodes with truthy relevant get deleted in getDataStrWithoutIrrelevantNodes

### ISSUE THREAD

**Issue by MartijnR** [enketo/enketo-core#443]
https://github.com/kobotoolbox/enketo-express/issues/803

**Comment by MartijnR** [comment]
Looks like the model is okay, but `getDataStr({irrelevant: false})` deletes the node when obtaining the stringified XML instance.

**Comment by MartijnR** [comment]
case of mistaken targeting. The node that should be removed is not removed and the node that should not be removed is removed.

```xml
<name_of_region>
            <project_focus_areas_livestock>
              <livestock_type>dairy_cattle</livestock_type>
              
            </project_focus_areas_livestock><project_focus_areas_livestock>
              <livestock_type>other</livestock_type>
              <livestock_type_other>one</livestock_type_other>
            </project_focus_areas_livestock>
          </name_of_region><name_of_region>
            <project_focus_areas_livestock>
              <livestock_type>beef_cattel</livestock_type>
              <livestock_type_other/> <!-- SHOULD HAVE BEEN REMOVED -->
            </project_focus_areas_livestock><project_focus_areas_livestock>
              <livestock_type>other</livestock_type>
              <!-- SHOULD NOT HAVE BEEN REMOVED -->
            </project_focus_areas_livestock>
          </name_of_region>
```

**Comment by MartijnR** [comment]
fixed by 320cf973ea49368404f536690853e9ba768457ed
